File: backend/Domain/Accommodation/Accommodation.cs
using Domain.Common.Base;

namespace Domain.Accommodation;
public class Accommodation : AuditableEntity
{
    public int Id { get; set; }
    public required string Name { get; set; }
    public string? Description { get; set; }
    public int? Rating { get; set; }
    public required string Street { get; set; }
    public required int StreetNumber { get; set; }
    public required string ZipCode { get; set; }
    public required string City { get; set; }
    public required string Region { get; set; }
    public required decimal Latitude { get; set; }
    public required decimal Longitude { get; set; }
    public required decimal PricePerNight { get; set; }
    public required int MaxGuests { get; set; }
    public virtual ICollection<AccommodationAmenity.AccommodationAmenity>? AccommodationAmenities { get; }
    public virtual ICollection<AccommodationImage.AccommodationImage>? AccommodationImages { get; }
    public virtual ICollection<Review.Review>? Reviews { get; }
    public virtual ICollection<Booking.Booking>? Bookings { get; }
}

File: frontend/ABMS/src/app/core/interfaces/accommodation.ts
import { ReviewDto } from '../../../api';

export interface Accommodation {
  id: number;
  name: string;
  description?: string;
  zipCode: string;
  region: string;
  street: string;
  streetNumber: number;
  city: string;
  latitude: number;
  longitude: number;
  unavailableCheckInDates: Date[];
  unavailableCheckOutDates: Date[];
  pricePerNight: number;
  maxGuests: number;
  images: string[];
  amenities: string[];
  reviews: ReviewDto[];
  isFavorite: boolean;
  rating?: number;
}
